projects/WTI-UI/src/app/modules/login/components/login-page/login-page.component.ts

```diff
@@ -124,17 +129,49 @@ export class LoginPageComponent implements OnInit, OnDestroy {
 			}
             this._contestService.isContestRunning = val;
             if (DEBUG_MODE) {
-            	console.log ("Invoking ContestService.contestClock.next()") ;
+            	console.log ("Invoking ContestService.contestClockEvent.next()") ;
             }
-            this._contestService.contestClock.next();
+			//trigger a contestClockEvent so the SelectProblems dropdown can decide whether to display the problems or not
+            this._contestService.contestClockEvent.next();
+
           });
-      }, (error: any) => {
-    	  if (DEBUG_MODE) {
-    		  console.log ("AuthService.login() subscription returned error.") ;
-    	  }
+
+		//get the actual contest clock info from the PC2 server via the Contest Service (which gets it via the WTI Server and its PC2 API)
+		if (DEBUG_MODE) {
+			console.log("  Invoking ContestService.getContestClock(), subscribing for callback result")
+		}
+		this._contestService.getContestClock() 
+			.subscribe(
+				(data: any) => {
+        			if (!data) { 
+						console.error ("LoginPageComponent.onSubmit() ContestClock subscription callback: unable to get ContestClock from PC2 API via ContestService!");
+					} else {						
+						//copy the data fields received from the PC2 Server (via the WTI-API) into the local ContestClock object
+						this.contestClock.isRunning = data.running ;
```

Comment:
There is a lot of nested conditionals here, maybe we can use break to break out of conditions. This would improve readability.

Reply:
I don't quite follow this comment.  As far as I can tell there are no nested conditionals which would allow using `break`.  What there ARE (or at least, WERE) are a lot of "debug if's".  I will remove all those debug statements; hopefully the logic will be clearer.

Reply:
I meant return sorry. Maybe the method can be terminated early if it is not going to lead anywhere?

Reply:
I've removed the large number of "debug" statements, which makes the nested conditionals pretty easy to follow (as JohnB has noted elsewhere, the traditional JavaScript usage of "lamba functions" does complicate it a little bit).   As it now stands I don't really see where there's any place that the addition of "return" statements would be particularly helpful...